### PR TITLE
CAMS-375 search by chapter

### PR DESCRIPTION
**Pull request by btposey**
# Purpose

Enable users to search for cases by chapter number

# Major Changes

* Added chapter number to filters form
* Added chapter number filter capability to API

# Testing/Validation

Test coverage above 90%.

## Definition of Done:
- [x] Acceptance criteria met - The system meets all acceptance criteria
- [x] Feature toggles created - Features that are deployed but not released have toggles
- [ ] Design QA passed - All visual UI looks and works the way it’s supposed to
- [ ] Usability validated - Team ensures the working code is easy to use
- [ ] Accessibility tests passed - System passes automated and manual accessibility tests and supports applicable devices
- [x] Code refactored for clarity - Developers can understand the work simply by reviewing the code
- [x] Dependency Rule followed - More important code doesn’t directly depend on less important code
- [x] Development debt eliminated - UX and code aligns to the team’s latest understanding of the domain
- [ ] Source code merged - Code has been merged into the main branch
- [x] 90% unit test coverage achieved - Automated test coverage tools indicate coverage of >= 90%
- [x] Code reviewed - Code is reviewed by at least two other team members before being merged
- [x] Code quality checks passed - Code passes all automated quality checks
- [x] Security scans passed - Code passes expected scans for vulnerabilities and compliance issues
- [x] Threat model updated - The threat model incorporates new threats and remediations 
- [x] Build process updated - Automated build process include code supporting the user story
- [x] Load/performance tests passed - Performance tests include the functionality of the user story
- [x] Documents are updated - All required documentation is up to date and version controlled
- [ ] Code is deployed - Code is deployed to the highest production like environment

### CODE REVIEW

File: user-interface/src/lib/components/CaseNumberInput.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ type CaseNumberInputProps = Omit<InputProps, 'onChange'> & {
 };
 
 function CaseNumberInputComponent(props: CaseNumberInputProps, ref: React.Ref<InputRef>) {
-  const { onChange, allowEnterKey, allowPartialCaseNumber } = props;
+  const { onChange, allowEnterKey, allowPartialCaseNumber, ...otherProps } = props;
```
Comment (amorrow-flexion) [inline user-interface/src/lib/components/CaseNumberInput.tsx:27]: I remember commenting something similar in another PR about using a spread of `...inputProps` versus `...otherProps`

File: backend/functions/lib/adapters/gateways/dxtr/cases.dxtr.gateway.ts [hunk 2]
```diff
@@ -290,12 +288,18 @@ export default class CasesDxtrGateway implements CasesInterface {
...
+    } else {
+      chapters.push('15');
+    }
 
-    const chapters: string[] = ['15'];
-    if (doChapter12Enable) chapters.push('12');
-    if (doChapter11Enable) chapters.push('11');
     parametersList.push(`cs.CS_CHAPTER IN ('${chapters.join("', '")}')`);
 
+    console.log(parametersList);
```
Comment (amorrow-flexion) [inline backend/functions/lib/adapters/gateways/dxtr/cases.dxtr.gateway.ts:302]: drop this log

File: backend/functions/lib/adapters/gateways/dxtr/cases.dxtr.gateway.ts [hunk 3]
```diff
@@ -269,13 +266,14 @@ export default class CasesDxtrGateway implements CasesInterface {
       value: predicate.offset ?? 0,
     });
 
+    console.log(`=============== case number ${predicate.caseNumber}`);
```
Comment (amorrow-flexion) [inline backend/functions/lib/adapters/gateways/dxtr/cases.dxtr.gateway.ts:269]: and this log

File: user-interface/src/search/SearchResults.test.tsx [hunk 4]
```diff
@@ -124,9 +124,35 @@ describe('SearchResults component tests', () => {
 
     await waitFor(() => {
       const pagination = document.querySelector('.usa-pagination');
+      expect(pagination).toBeInTheDocument();
+      expect(pagination).toBeVisible();
+    });
 
+    // TODO: click the next page...
```
Comment (btposey) [inline user-interface/src/search/SearchResults.test.tsx:131]: Remove TODO.

**Review by fmaddenflx — APPROVED** [review]
Looks good.

**Review by amorrow-flexion — APPROVED** [review]

**Pull request merged**